### PR TITLE
DOC-167 added the method/condition to evaluate the user existence for…

### PR DESCRIPTION
**Pull request by mani0201**
… Create/Delete api's

### CODE REVIEW

File: datasafe-rest-impl/src/main/java/de/adorsys/datasafe/rest/impl/controller/UserController.java [hunk 1]
```diff
@@ -41,7 +43,10 @@ public void createUser(@RequestBody UserDTO requestDTO) {
         ReadKeyPassword readKeyPassword = new ReadKeyPassword(requestDTO.getPassword());
         UserIDAuth auth = new UserIDAuth(new UserID(requestDTO.getUserName()), readKeyPassword);
 
-        dataSafeService.userProfile().registerUsingDefaults(auth);
+        if (dataSafeService.userProfile().userExists(auth.getUserID())) {
+            throw new RestImplException("user \"" + auth.getUserID().getValue() + "\" already exists");
+        }else
```
Comment (valb3r) [inline datasafe-rest-impl/src/main/java/de/adorsys/datasafe/rest/impl/controller/UserController.java]: There is no need for else block here

File: datasafe-rest-impl/src/main/java/de/adorsys/datasafe/rest/impl/controller/UserController.java [hunk 2]
```diff
@@ -41,7 +43,10 @@ public void createUser(@RequestBody UserDTO requestDTO) {
         ReadKeyPassword readKeyPassword = new ReadKeyPassword(requestDTO.getPassword());
         UserIDAuth auth = new UserIDAuth(new UserID(requestDTO.getUserName()), readKeyPassword);
 
-        dataSafeService.userProfile().registerUsingDefaults(auth);
+        if (dataSafeService.userProfile().userExists(auth.getUserID())) {
+            throw new RestImplException("user \"" + auth.getUserID().getValue() + "\" already exists");
```
Comment (valb3r) [inline datasafe-rest-impl/src/main/java/de/adorsys/datasafe/rest/impl/controller/UserController.java]: Makes sense to throw more specific exception like UserExistsException

File: datasafe-rest-impl/src/main/java/de/adorsys/datasafe/rest/impl/controller/UserController.java [hunk 3]
```diff
@@ -53,7 +58,12 @@ public void createUser(@RequestBody UserDTO requestDTO) {
     @ApiOperation("Delete user")
     public void deleteUser(@RequestHeader String user,
                            @RequestHeader String password) {
-        UserIDAuth userIDAuth = new UserIDAuth(new UserID(user), new ReadKeyPassword(password));
-        dataSafeService.userProfile().deregister(userIDAuth);
+        UserIDAuth auth = new UserIDAuth(new UserID(user), new ReadKeyPassword(password));
+
+        if (dataSafeService.userProfile().userExists(auth.getUserID()))
+            dataSafeService.userProfile().deregister(auth);
+        else
```
Comment (valb3r) [inline datasafe-rest-impl/src/main/java/de/adorsys/datasafe/rest/impl/controller/UserController.java]: Makes sense to invert check (to reduce nestedness), also please use if with curly braces (see https://stackoverflow.com/questions/2125066/is-it-a-bad-practice-to-use-an-if-statement-without-curly-braces):
```
if (!dataSafeService.userProfile().userExists(auth.getUserID())) {
    throw new UserDoesNotExists("User \"" + auth.getUserID().getValue() + "\" does not exists");
}

dataSafeService.userProfile().deregister(auth);
```

File: datasafe-rest-impl/src/main/java/de/adorsys/datasafe/rest/impl/exceptions/RestImplException.java [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package de.adorsys.datasafe.rest.impl.exceptions;
+
+public class RestImplException extends RuntimeException {
```
Comment (valb3r) [inline datasafe-rest-impl/src/main/java/de/adorsys/datasafe/rest/impl/exceptions/RestImplException.java]: Better to have 2 separate exceptions - UserDoesNotExists,UserExistsException

**Comment by valb3r** [comment]
Also, please don't forget to add tests to UserControllerTest

File: datasafe-rest-impl/src/main/java/de/adorsys/datasafe/rest/impl/controller/UserController.java (hunk 1 above)
Comment (mani0201) [reply datasafe-rest-impl/src/main/java/de/adorsys/datasafe/rest/impl/controller/UserController.java]: ok.

File: datasafe-rest-impl/src/main/java/de/adorsys/datasafe/rest/impl/controller/UserController.java (hunk 2 above)
Comment (mani0201) [reply datasafe-rest-impl/src/main/java/de/adorsys/datasafe/rest/impl/controller/UserController.java]: ok.

File: datasafe-rest-impl/src/main/java/de/adorsys/datasafe/rest/impl/controller/UserController.java (hunk 3 above)
Comment (mani0201) [reply datasafe-rest-impl/src/main/java/de/adorsys/datasafe/rest/impl/controller/UserController.java]: ok.

File: datasafe-rest-impl/src/main/java/de/adorsys/datasafe/rest/impl/exceptions/RestImplException.java (hunk 4 above)
Comment (mani0201) [reply datasafe-rest-impl/src/main/java/de/adorsys/datasafe/rest/impl/exceptions/RestImplException.java]: ok.

**Comment by mani0201** [comment]
updated based on review.

**Pull request closed without merging**